media-sound/mpd: move dev-libs/boost to CDEPEND

dset0x opened the pull request.

SoapGentoo commented on media-sound/mpd/mpd-0.19.21-r1.ebuild:
```diff
@@ -0,0 +1,258 @@
+	local mpdconf="--enable-database --disable-roar --disable-documentation
+		--enable-dsd --enable-largefile --disable-osx --disable-shine-encoder
+		--disable-solaris-output --enable-tcp --enable-un --disable-werror
+		--docdir=${EPREFIX}/usr/share/doc/${PF}"
```
don't need to specify `--docdir` in `EAPI 6`

SoapGentoo commented on media-sound/mpd/mpd-0.19.21-r1.ebuild:
```diff
@@ -0,0 +1,258 @@
+}
+
+src_install() {
+	emake DESTDIR="${D}" install
```
`default` instead

SoapGentoo commented on media-sound/mpd/mpd-0.19.21-r1.ebuild:
```diff
@@ -0,0 +1,257 @@
+	zip? ( dev-libs/zziplib )
+	zlib? ( sys-libs/zlib )"
+DEPEND="${CDEPEND}
+	dev-libs/boost
```
move boost to `CDEPEND` and depend on it with `:=`

dset0x commented:
Thank you for all the tips @SoapGentoo. Here's the diff for the init file.

```diff
--- mpd-0.19.21.ebuild	2016-12-23 17:55:25.380021325 +0200
+++ mpd-0.19.21-r1.ebuild	2017-01-04 23:45:18.270008592 +0200
@@ -41,7 +41,8 @@
 	vorbis? ( glib )
 	wavpack? ( glib )"
 
-CDEPEND="!<sys-cluster/mpich2-1.4_rc2
+CDEPEND="dev-libs/boost:=
+	!<sys-cluster/mpich2-1.4_rc2
 	adplug? ( media-libs/adplug )
 	alsa? ( media-sound/alsa-utils
 		media-libs/alsa-lib )
@@ -96,7 +97,6 @@
 	zip? ( dev-libs/zziplib )
 	zlib? ( sys-libs/zlib )"
 DEPEND="${CDEPEND}
-	dev-libs/boost
 	virtual/pkgconfig"
 RDEPEND="${CDEPEND}
 	selinux? ( sec-policy/selinux-mpd )
@@ -142,8 +142,7 @@
 src_configure() {
 	local mpdconf="--enable-database --disable-roar --disable-documentation
 		--enable-dsd --enable-largefile --disable-osx --disable-shine-encoder
-		--disable-solaris-output --enable-tcp --enable-un --disable-werror
-		--docdir=${EPREFIX}/usr/share/doc/${PF}"
+		--disable-solaris-output --enable-tcp --enable-un --disable-werror"
 
 	if use network; then
 		mpdconf+=" --enable-shout $(use_enable vorbis vorbis-encoder)
@@ -223,12 +222,13 @@
 }
 
 src_install() {
-	emake DESTDIR="${D}" install
+	default
 
 	insinto /etc
 	newins doc/mpdconf.dist mpd.conf
 
-	newinitd "${FILESDIR}"/${PN}2.init ${PN}
+	newconfd "${FILESDIR}"/${PN}-0.19.21.conf ${PN}
+	newinitd "${FILESDIR}"/${PN}-0.19.21.init ${PN}
 
 	systemd_newuserunit systemd/system/${PN}.service ${PN}.service
 	sed -i '/WantedBy=/c WantedBy=default.target' \
```

dset0x commented:
Please don't review yet. I need to study the new init file first.

dset0x commented:
Looks like my work is obsolete, so I've reduced this to the smaller fixes it contained.

```diff
--- mpd-0.20.4.ebuild	2017-03-15 10:36:32.050012946 +0200
+++ mpd-0.20.4-r1.ebuild	2017-03-15 12:03:39.410019628 +0200
@@ -40,7 +40,8 @@
 	vorbis? ( glib )
 	wavpack? ( glib )"
 
-CDEPEND="!<sys-cluster/mpich2-1.4_rc2
+CDEPEND="dev-libs/boost:=
+	!<sys-cluster/mpich2-1.4_rc2
 	adplug? ( media-libs/adplug )
 	alsa? ( media-sound/alsa-utils
 		media-libs/alsa-lib )
@@ -95,7 +96,6 @@
 	zip? ( dev-libs/zziplib )
 	zlib? ( sys-libs/zlib )"
 DEPEND="${CDEPEND}
-	dev-libs/boost
 	virtual/pkgconfig"
 RDEPEND="${CDEPEND}
 	selinux? ( sec-policy/selinux-mpd )
@@ -223,7 +223,7 @@
 }
 
 src_install() {
-	emake DESTDIR="${D}" install
+	default
 
 	insinto /etc
 	newins doc/mpdconf.dist mpd.conf
```

a17r commented:
lgtm, it seems you could also drop eutils.eclass.

Meanwhile there has been a 0.20.6 release, are you motivated to combine that with the EAPI 6 bump? ;)

dset0x commented:
@a17r I went ahead and renamed to 0.20.6 since it's a bugfix release.

The recent init file contains the following file however,
```
command_args=${opts:=${CFGFILE}}
```

openrc-run complains,
```
 * Use of the opts variable is deprecated and will be
 * removed in the future.
 * Please use extra_commands, extra_started_commands or extra_stopped_commands.
```

I assume it's there to be able to specify different config files, but I don't think openrc is meant to be used in this dynamic way. Should we drop it here?

mgorny commented:
I'm sorry but it seems that this pull request got outdated again. Could you rebase it and possibly update the description? Because it seems that it's gotten past simple dep update.

mgorny commented:
Closing due to no activity. Feel free to open a new pull request once you rebase and update the patch (if it is still necessary).
